### PR TITLE
Product type issue fix

### PR DESCRIPTION
**Pull request by godrei**

### CODE REVIEW

File: xcodeproj/target_test.go [hunk 1]
```diff
@@ -200,7 +199,7 @@ const expectedAggregateTarget = `{
 	"ProductReference": {
 		"Path": ""
 	},
-	"ProductType": "com.apple.product-type.application"
```
Comment (trapacska) [inline xcodeproj/target_test.go]: as it is not required it doesn't mean it cannot be "com.apple.product-type.application" as well, now it is checked for "" only

File: xcworkspace/errors.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import "fmt"
+
+// SchemeNotFoundError represents that the given scheme was not found in the container
+type SchemeNotFoundError struct {
+	scheme    string
+	container string
+}
+
+// NewSchemeNotFoundError creates a new SchemeNotFoundError
+func NewSchemeNotFoundError(scheme, container string) SchemeNotFoundError {
```
Comment (trapacska) [inline xcworkspace/errors.go]: I think the new function is not essential for error interface matcher classes. The biggest issue I have with it is that the function name contains the whole return type name. Also it doesn't optimizes the length too much. I won't mix xcworkspace's functions that starts with `New` with an error creating function.

I used these artifacts to validate my thinking:
https://yourbasic.org/golang/create-error/
http://pro-tips-dot-com.tumblr.com/post/56007335165/custom-error-types-in-go
https://programming.guide/go/create-error.html
https://golangbot.com/custom-errors/
https://gobyexample.com/errors

File: xcworkspace/errors_test.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
+package xcworkspace
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+)
+
+func TestNewSchemeNotFoundError(t *testing.T) {
```
Comment (trapacska) [inline xcworkspace/errors_test.go]: we won't need this kind of unit tests if we don't have that New.. function

File: xcworkspace/errors_test.go [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			e := SchemeNotFoundError{
+				scheme:    tt.scheme,
+				container: tt.container,
+			}
+			if got := e.Error(); got != tt.want {
+				t.Errorf("SchemeNotFoundError.Error() = %v, want %v", got, tt.want)
+			}
```
Comment (trapacska) [inline xcworkspace/errors_test.go]: I would also test err type

File: xcworkspace/errors_test.go [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+		scheme    string
+		container string
+		want      string
+	}{
+		{
+			name:      "simple test",
+			scheme:    "Scheme",
+			container: "Workspace",
+			want:      "scheme Scheme not found in Workspace",
+		},
```
Comment (trapacska) [inline xcworkspace/errors_test.go]: you have TBT but you've set only one test sample data

Comment (trapacska) [reply xcworkspace/errors_test.go]: https://github.com/golang/go/wiki/TableDrivenTests

File: xcodeproj/target_test.go (hunk 1 above)
Comment (godrei) [reply xcodeproj/target_test.go]: we have test case for that as well: rawLegacyTarget and rawNativeTarget

**Pull request merged**